### PR TITLE
Scalability

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

File: roadblock.py [hunk 1]
```diff
@@ -215,6 +213,13 @@ def set_message_validation(self, msg_val):
 
         return 0
 
+    def set_connection_watchdog(self, con_watchdog_state):
+        '''Enable/disable the connection watchdog'''
+
+        self.connection_watchdog_state = con_watchdog_state
```
Comment (rafaelfolco) [inline roadblock.py:219]: You must have a good reason for creating this function. 
If you are creating a function only for a variable assignment, you could also do one or more of the following:
- get the actual var value 
- print debug info with the var value
- check if value is valid
- handle possible errors or unexpected situations

File: roadblock.py [hunk 2]
```diff
@@ -976,16 +972,25 @@ def message_publish(self, message):
...
             counter += 1
-            # this call should return the number of clients that receive the message
-            # we expect it to be greater than zero, if not we retry
-            ret_val = self.redcon.publish(self.roadblock_uuid + "__busB", message_str)
 
-            if ret_val == 0:
+            try:
+                # need to figure out the foo:bar thing...it feels like
+                # we don't really need information there but it
+                # appears to be a required field :/
```
Comment (rafaelfolco) [inline roadblock.py:979]: perhaps you are creating the bus without the required fields so it does that by default?

File: roadblocker.py [hunk 3]
```diff
@@ -126,55 +118,62 @@ def process_options ():
                         help = argparse.SUPPRESS,
                         action = "store_true")
 
-    t_global.args = parser.parse_args()
+    args = parser.parse_args()
```
Comment (rafaelfolco) [inline roadblocker.py:121]: this makes me feel better

File: roadblocker.py [hunk 4]
```diff
@@ -126,55 +118,62 @@ def process_options ():
...
 
-    process_options()
+    args = process_options()
+
+    # log formatting variables
+    log_debug_format =  '[CODE][%(module)s %(funcName)s:%(lineno)d]\n[%(asctime)s][%(levelname) 8s] %(message)s'
+    log_normal_format = '[%(asctime)s][%(levelname) 8s] %(message)s'
+
+    if args.log_level == 'debug':
+        logging.basicConfig(level = logging.DEBUG, format = log_debug_format, stream = sys.stdout)
```
Comment (rafaelfolco) [inline roadblocker.py:147]: you have debug logging, use it in that function :)

**Review by rafaelfolco — APPROVED** [review]
lgtm, a few questions/suggestions in case of a ci/change respin.

File: roadblock.py (hunk 2 above)
Comment (k-rister) [reply roadblock.py]: This was one thing I wanted to resolve and I had forgotten about it.  I just figured this out and removed it.

**Pull request merged**